### PR TITLE
feat(portal2)!: modified playbook to support portal2

### PR DESCRIPTION
**Pull request by klyse**
BREAKING CHANGE: portal2 support in the playbook

### CODE REVIEW

**Comment by klyse** [comment]
The unit tests are not really ready yet. The code should be tho :)

**Comment by klyse** [comment]
and I'm unsure what's going on on the with the tests..

**Comment by burnedikt** [comment]
> and I'm unsure what's going on on the with the tests..

Seems to be this issue:

```
Error creating container: 400 Client Error for http+docker://localhost/v1.41/containers/create?name=molecule-docker%24%7BINSTANCE_NAME_SUFFIX%7D: Bad Request (\"Invalid container name (molecule-docker${INSTANCE_NAME_SUFFIX}), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed\")
```

Seems like `INSTANCE_NAME_SUFFIX` is not set

**Comment by burnedikt** [comment]
I'll add some more commits here to remove support for older backend versions as well so we have a clean role for Portal v2, Cloud Rendering and Backend v4

**Comment by burnedikt** [comment]
> and I'm unsure what's going on on the with the tests..

fixed with https://github.com/Innoactive/ansible-innoactive-portal-role/pull/118/commits/14f0509da7614be32bd87757d6f23260e90a584f

File: defaults/main.yml [hunk 1]
```diff
@@ -58,15 +58,19 @@ traefik:
...
+session_management_image_version: latest
+cloudxr_management_image_version: latest
 
 mailserver_image: namshi/smtp:latest
 message_queue_image: redis:5
 database_image: postgres:9.6
 # legacy parameter to specify portal backend image is hub_image. Defaults to registry.docker.innoactive.de/portal_backend:latest
 portal_backend_image: registry.docker.innoactive.de/{{ hub_image | default('portal_backend') }}:{{ portal_backend_image_version }}
-portal_image: registry.docker.innoactive.de/hub_portal_web:{{ portal_image_version }}
+portal_image: registry.docker.innoactive.de/portal_frontend:{{ portal_image_version }}
```
Comment (klyse) [inline defaults/main.yml:69]: this is probably the wrong image

Comment (burnedikt) [reply defaults/main.yml:69]: why?

Comment (klyse) [reply defaults/main.yml:69]: isn't this the backend container var? if not than I got it confused with the `hub_web` image..

Comment (burnedikt) [reply defaults/main.yml:69]: `portal_image` is the portal frontend, `portal_backend_image` is the portal backend (f.k.a `hub_image`)

**Comment by burnedikt** [comment]
Overview of changes

### Breaking Changes

- Dropped Support for Portal Backend Versions < 4.x
  - consequently also dropped support for `hub_offering`, `concurrent_access_tokens` variables and JWT features
- Dropped Support for Portal Frontend < 2.x
  - Removed `portal_oauth_client_secret` and `portal_enabled_features` variables
- Renamed everything related to "Discovery Portal" to "Portal Frontend"
  - Ansible tag `discovery_portal` replaced by `frontend`

### Features

- added support to deploy CloudXR Management service
- added support to deploy Session Management service
- added support to deploy Signalling Proxy (proxy WSS to WS for render nodes)

### Deprecations

- variable `setup_wmc` has been superseded by `setup_control_panel`
- variable `setup_discovery_portal` superseded by `setup_portal_frontend`
- variable `launcher_oauth_client_id` superseded by `desktop_client_oauth2_client_id`
- variable `launcher_oauth_client_secret` superseded by `desktop_client_oauth2_client_secret`

**Comment by burnedikt** [comment]
@klyse I've added the breaking changes because of Backend Version 4. Tests are also fixed. However, we now have an issue related to setting up cloudxr (see CI), maybe you can take a look?

```json
{"msg": "The conditional check 'cloudxr_configuration.enabled' failed. The error was: error while evaluating conditional (cloudxr_configuration.enabled): {'enabled': '{{ cloudxr_enabled }}', 'defaultRegion': '{{ cloudxr_default_region }}', 'instance': '{{ cloudxr_instance | default(instance_name, true) }}', 'max_cloud_instances': '{{ cloudxr_max_cloud_instances }}', 'max_keep_avlive_age': '{{ cloudxr_machine_max_keep_alive_age }}', 'sentry_dsn': '{{ cloudxr_management_sentry_dsn }}', 'ip_stack_api_token': '{{ cloudxr_ip_stack_api_token }}', 'azure_enabled': \"{{ cloudxr_azure_enabled | default('false', true) }}\", 'azure_client_id': \"{{ cloudxr_azure_client_id | default('false', true) }}\", 'azure_client_secret': \"{{ cloudxr_azure_secret | default('false', true) }}\", 'azure_subscription': \"{{ cloudxr_azure_subscription | default('false', true) }}\", 'azure_tenant_id': \"{{ cloudxr_azure_tenant_id | default('false', true) }}\", 'aws_enabled': \"{{ cloudxr_aws_enabled | default('false', true) }}\", 'aws_access_key_id': \"{{ cloudxr_aws_access_key_id | default('', true) }}\", 'aws_secret_access_key': \"{{ cloudxr_aws_secret_access_key | default('', true) }}\", 'aws_regions': \"{{ cloudxr_aws_regions | default('', true) }}\"}: 'instance_name' is undefined\n\nThe error appears to be in '/home/vsts/work/1/s/tasks/cloudxr_management_service.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: Start CloudXR management container\n  ^ here\n"}
```

**Comment by klyse** [comment]
The issue was that the variable `instance_name` exists only in our configuration management project and I was using that as a fallback if the `cloudxr_instance` is not set. Should be fixed now.

**Review by burnedikt — APPROVED** [review]

**Pull request merged**